test/docs/transactions.test.js

```diff
@@ -396,4 +396,29 @@ describe('transactions', function() {
     assert.equal(docs.length, 1);
     assert.equal(docs[0].name, 'test');
   });
+
+  it('transaction() retains modified status for documents created outside the transaction (gh-13973)', async function() {
+    db.deleteModel(/Test/);
+    const Test = db.model('Test', Schema({ status: String }));
+
+    await Test.createCollection();
+    await Test.deleteMany({});
+
+    const { _id } = await Test.create({ status: 'test' });
+    const doc = await Test.findById(_id);
+
+    let i = 0;
+    await db.transaction(async(session) => {
+      doc.status = 'test2';
+      assert.ok(doc.$isModified('status'));
+      await doc.save({ session });
```

Comment:
from what i can tell, this test tests modified status *within* a transcation, but the title makes out sound like retaining modified status across transaction if modified *outside* the transaction, like (pseudo):

```js
const doc = await Model.find();
doc.test = "something"; // not saved outside of transaction
await db.transaction(async(session) => { // run at least twice, once with error
  assert.ok(doc.$isModified('test'));
  await doc.save({ session });
  throw new Error();
});
```

Reply:
test title was changed, so it makes it more clear what it is testing, thanks